cloud/google/gcdns_record.py

```diff
@@ -0,0 +1,789 @@
+        if record_type == 'NS' and record_name[0] == '*':
+            module.fail_json(
+                msg     = "wildcard NS records not allowed, got: %s" % record_name,
+                changed = False
+            )
+
+        # Values for txt records must begin and end with a double quote.
+        if record_type == 'TXT':
+            for value in values:
+                if value[0] != '"' and value[-1] != '"':
+                    module.fail_json(
+                        msg     = 'TXT values must be enclosed in double quotes, ' +
+                                  'got: %s' % value,
+                        changed = False
+                    )
+
+
+def _additional_sanity_checks(module, zone):
+    """Run input sanity checks that depend on info from the zone/record."""
+
+    overwrite   = module.params['overwrite']
+    record_name = module.params['record']
+    record_type = module.params['type']
+    state       = module.params['state']
+
+    # CNAME records are not allowed to have the same name as the root domain.
+    if record_type == 'CNAME' and record_name == zone.domain:
+        module.fail_json(
+            msg     = 'CNAME records cannot match the zone name',
+            changed = False
+        )
+
+    # The root domain must always have an NS record.
+    if record_type == 'NS' and record_name == zone.domain and state == 'absent':
+        module.fail_json(
+            msg     = 'cannot delete root NS records',
+            changed = False
+        )
+
+    # Updating NS records with the name as the root domain is not allowed
+    # because libcloud does not support in-place updates and root domain NS
+    # records cannot be removed.
+    if record_type == 'NS' and record_name == zone.domain and overwrite:
+        module.fail_json(
+            msg     = 'cannot update existing root NS records',
+            changed = False
+        )
+
+    # SOA records with names that don't match the root domain are not permitted
+    # (and wouldn't make sense anyway).
+    if record_type == 'SOA' and record_name != zone.domain:
+        module.fail_json(
+            msg     = 'non-root SOA records are not permitted, got: %s' % record_name,
+            changed = False
+        )
+
+
+################################################################################
+# Main
+################################################################################
+
+def main():
+    """Main function"""
+
+    module = AnsibleModule(
+        argument_spec = dict(
+            state                 = dict(default='present', choices=['present', 'absent'], type='str'),
+            record                = dict(required=True, aliases=['name'], type='str'),
+            zone                  = dict(type='str'),
+            zone_id               = dict(type='str'),
+            type                  = dict(required=True, choices=SUPPORTED_RECORD_TYPES, type='str'),
+            values                = dict(type='list'),
```

Comment:
even though this is a list, you can also use it as a single str item. therefore this works:

~~~
- gcdns_record:
   record: 'www.example.com'
   zone_id: 'example-com'
   type: CNAME
   values: www.example.com.    # Note the trailing dot
~~~
But values are in plural, To make it more readable I would recommend to add an alias `aliases=['value']` 

~~~
- gcdns_record:
   record: 'www.example.com'
   zone_id: 'example-com'
   type: CNAME
   value: www.example.com.    # Note the trailing dot
~~~